scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -141,6 +142,95 @@ object DynamicTLSPeerGroup {
     }
   }
 
+  sealed abstract case class FramingConfig private (
+      maxFrameLength: Int,
+      lengthFieldOffset: Int,
+      lengthFieldLength: Int,
+      encodingLengthAdjustment: Int,
+      decodingLengthAdjustment: Int,
+      initialBytesToStrip: Int,
+      failFast: Boolean,
+      byteOrder: ByteOrder,
+      lengthIncludesLengthFieldLength: Boolean
+  )
+
+  object FramingConfig {
+    private val possibleLengthFieldLength = Set(1, 2, 3, 4, 8)
+
+    /**
+      *
+      * Configures framing format for all the peers in PeerGroup
+      *
+      *
+      * @param maxFrameLength the maximum length of the frame. If the length of the frame is greater than this value
+      *                       channel with generate DecodingError event.
```

Comment:
So first question depends on `lengthIncludesLengthFieldLength` flag  i.e. if it is `true` then it is true that prepended length is part of the `maxFrameLength`.

And second question depends on `failFast` flag. if its `true` then it is happening as you described if `false`  then the whole thing is read and then checked

Reply:
Hm, then I may have misunderstood `lengthIncludesLengthFieldLength`. 

What I thought it meant was that for example:
1. I configure a `lengthFieldLength` of 4
2. I'm reading the stream of data
3. I read 4 bytes
4. I evaluate those bytes to `length`
5. Because `lengthIncludesLengthFieldLength` is true, I assume it will be 4 higher than what the length of the data that follows.
6. I read `length - 4` bytes

Whereas if `lengthIncludesLenghtFieldLength` was false, I'd assume that it goes:
...
5. Because `lengthIncludesLengthFieldLength` is false, I assume it will be only contain the length of the actual content
6. I read `length` bytes

I thought that in both cases, the `maxFrameLength` I have to configure has to cater for the data as well as the prefix.

Reply:
oh maybe I have said somehing not precise enough. So in case of option `lengthIncludesLengthFieldLength` it works only on `encoder` i.e it is used only in `LengthFieldPrepender` class, and when `true` length prefix is increased by length prefix length. 

On decoder side i.e in `LengthFieldBasedFrameDecoder`, in that case one need to configure option `decodingLengthAdjustment` to be inverse of length field length to properly account from it with `maxFrameLength`
from netty docs:
```
In most cases, the length field represents the length of the message body only, as shown in the previous examples. However, in some protocols, the length field represents the length of the whole message, including the message header. In such a case, we specify a non-zero lengthAdjustment. Because the length value in this example message is always greater than the body length by 2, we specify -2 as lengthAdjustment for compensation.
   lengthFieldOffset   =  0
   lengthFieldLength   =  2
   lengthAdjustment    = -2 (= the length of the Length field)
   initialBytesToStrip =  0
```

Reply:
I tap out :upside_down_face: 

I put all my hope that the standard use case is simple and that's all anyone ever has to use, or else take a deep dive in Netty docs.